protocol/daemons/types/health_checkable.go

```diff
@@ -0,0 +1,113 @@
+package types
+
+import (
+	"fmt"
+	libtime "github.com/dydxprotocol/v4-chain/protocol/lib/time"
+	"sync"
+	"time"
+)
+
+const (
+	MaximumAcceptableUpdateDelay = 5 * time.Minute
+)
+
+// HealthCheckable is a common interface for services that can be health checked.
+type HealthCheckable interface {
+	// HealthCheck returns an error if a service is unhealthy. If the service is healthy, this method returns nil.
+	HealthCheck(provider libtime.TimeProvider) (err error)
+}
+
+// HealthCheckableImpl implements the HealthCheckable interface by tracking the timestamps of the last successful and
+// failed updates. If:
+// - no update has occurred
+// - the most recent update failed, or
+// - the daemon has not seen a successful update in at least 5 minutes,
+// the service is considered unhealthy.
+type HealthCheckableImpl struct {
```

Comment:
don't expose this, instead have the interface expose what methods you want like
`HealthCheck` and `Report*` and have `healthCheckImpl` implement them. This way the internal implementation doesn't matter.

This also allows you to merge instantiation (via `NewHealthCheckableImpl`) and initialization (via `InitializeHealthStatus`) into a single method that instantiates and initializes.

If you don't want to expose the `Report*` on the `HealthCheckable` then you could have two interfaces, one which is the `HealthCheckable` and `HealthReporter` that splits them up but have `HealthCheckableImpl` implement both of them.